tests/conftest.py

```diff
@@ -0,0 +1,137 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+# -----------------------------------------------------------------------------
+"""Testing framework infrastructure."""
+
+import os
+from pathlib import Path
+
+import pytest
+from _pytest.config import Config
+from _pytest.config.argparsing import Parser
+
+
+def pytest_addoption(parser: Parser) -> None:
+    """Add custom command line options to pytest.
+
+    Args:
+        parser: Pytest parser
+    """
+
+    parser.addoption(
+        "--4C_ArborX",
+        action="store_true",
+        default=False,
+        help="Execute 4C and ArborX based tests.",
+    )
+
+    parser.addoption(
+        "--CubitPy",
+        action="store_true",
+        default=False,
+        help="Execute CubitPy based tests.",
+    )
+
+    parser.addoption(
+        "--performance",
+        action="store_true",
+        default=False,
+        help="Execute performance tests.",
+    )
+
+
+def pytest_collection_modifyitems(config: Config, items: list) -> None:
+    """Filter tests based on their markers and provided command line options.
+
+    Currently configured options:
+        `pytest`: execute tests with no markers
+        `pytest --4C_ArborX`: execute tests with the `fourc_arborx` marker
+        `pytest --CubitPy`: execute tests with the `cubitpy` marker
+        `pytest --performance`: execute tests with the `performance` marker
+
+    Args:
+        config: Pytest config
+        items: Pytest list of tests
+    """
+
+    selected_tests = []
+
+    # loop over all collected tests
+    for item in items:
+        # Get all set markers for current test (e.g. `fourc_arborx`, `cubitpy`, `performance`)
+        markers = [marker.name for marker in item.iter_markers()]
+
+        if config.getoption("--4C_ArborX") and "fourc_arborx" in markers:
+            selected_tests.append(item)
+
+        if config.getoption("--CubitPy") and "cubitpy" in markers:
+            selected_tests.append(item)
+
+        if config.getoption("--performance") and "performance" in markers:
+            selected_tests.append(item)
+
+        if not markers:
+            selected_tests.append(item)
+
+    deselected_tests = list(set(items) - set(selected_tests))
+
+    items[:] = selected_tests
+    config.hook.pytest_deselected(items=deselected_tests)
+
+
+@pytest.fixture(scope="session")
+def reference_file_directory() -> Path:
+    """Provide the path to the reference file directory.
+
+    Returns:
+        Path: A Path object representing the full path to the reference file directory.
+    """
+
+    testing_path = os.path.abspath(os.path.dirname(__file__))
+    return Path(testing_path) / "reference-files"
+
+
+@pytest.fixture(scope="function")
+def current_test_name(request: pytest.FixtureRequest) -> str:
```

Comment:
Is it really necessary to provide the argument `requests` here? I am not sure if we have this available everywhere where we need to call this function. Maybe (for the beginning?) we should stick to the `env` variable approach. What do you think?

Reply:
I think https://stackoverflow.com/questions/17726954/py-test-how-to-get-the-current-tests-name-from-the-setup-method sums it up really nicely

This will also work for parameterized tests (we said we probably want those) and also it works for test functions as well as test methods.

The argument requests is always available once pytest is available.

Note: We do not need to provide the argument request. We simply call it via `current_test_name`.

Reply:
> Note: We do not need to provide the argument request. We simply call it via current_test_name.

Sorry, I was not aware of that, then this is perfectly fine the way it is.